src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -877,10 +878,9 @@ private void ScanSyntaxToken(ref TokenInfo info)
                     if (_badTokenCount++ > 200)
```

Comment:
I think this is safe. If the magic number is changed upwards, this test will fail, as the number of tokens and the value of the last token won't match expectations anymore.